src/command/process_update_task.rs

```diff
@@ -162,46 +293,239 @@ impl UpdateProcessor {
+                    .await?;
+
                 self.return_to_initial().await?;
 
-                self.pattern_response(number).await
+                self.get_weather(city).await
             }
 
             Err(_) => self.not_number_message().await,
         }
     }
 
-    async fn pattern_response(&self, number: usize) -> Result<(), BotError> {
-        let city = self
-            .repo
-            .get_city_row(&self.chat.selected.clone().unwrap(), number)
+    async fn process_set_city_number(&self) -> Result<(), BotError> {
+        match self.text.parse::<usize>() {
+            Ok(number) => {
+                let city = self
+                    .repo
+                    .get_city_row(&self.chat.selected.clone().unwrap(), number)
+                    .await?;
+
+                self.return_to_initial().await?;
+
+                self.set_default_city(city).await
+            }
+
+            Err(_) => self.not_number_message().await,
+        }
+    }
+
+    async fn not_valid_offset_message(&self) -> Result<(), BotError> {
+        self.cancel(Some(
+            "That's not a valid offset, it has to be a number in range [-11, 12].\n
+            If your timezone is UTC + 2 put 2, if you have UTC - 10 put -10, 0 if you have UTC timezone.\n
+            The command was cancelled"
+            .to_string(),
+        ))
+        .await?;
+
+        Ok(())
+    }
+
+    async fn schedule_forecast(
+        &self,
+        offset: i8,
+        city_id: i32,
+        user_hour: i8,
+        minutes: i8,
+    ) -> Result<(), BotError> {
+        let hour_utc = if user_hour - offset < 0 {
+            user_hour - offset + 24
+        } else {
+            user_hour - offset
+        };
+
+        let cron_expression = format!("0 {} {} * * * *", minutes, hour_utc);
+
+        let datetime = Repo::calculate_next_delivery(&cron_expression)?;
+
+        // Here we should call repo.insert_forecast
+        // We have to ask city_id for now default city id set
+        self.repo
+            .update_or_insert_forecast(
+                &self.chat.id,
+                self.chat.user_id,
+                &city_id,
+                cron_expression,
+                datetime,
+            )
             .await?;
 
-        match self.chat.before_state {
-            ClientState::Initial => self.get_weather(city).await,
+        self.return_to_initial().await?;
+
+        let minutes_pretty = if minutes < 10 {
+            format!("0{}", minutes)
+        } else {
+            minutes.to_string()
+        };
+
+        let text = format!(
+            "Weather info scheduled every day at {}:{} UTC {}",
+            user_hour, minutes_pretty, offset
+        );
+
+        self.send_message(&text).await
+    }
+
+    async fn process_offset(&self) -> Result<(), BotError> {
+        match self.text.parse::<i8>() {
+            Ok(offset) => {
+                if !(-11..=12).contains(&offset) {
```

Comment:
I think it can be simplified -11 < offset && offset < 12

Reply:
Clippy is special with this.